corehq/apps/locations/models.py

```diff
@@ -430,11 +430,24 @@ def mptt_get_descendants(self, **kw):
         return super(SQLLocation, self).get_descendants(**kw)
 
     def get_ancestors(self, **kw):
+        """
+        Returns list of all ancestor locations of this location
+        """
         timing = TimingContext("get_ancestors")
         with timing("mptt"):
             mptt_set = self.mptt_get_ancestors(**kw)
         return ComparedQuerySet(mptt_set, timing)
 
+    def get_ancestor_of_type(self, type_code):
+        """
+        Returns the ancestor of given location_type_code of the location
+        """
+        for loc in self.get_ancestors():
+            if loc.location_type.code == type_code:
+                return loc
+        raise SQLLocation.DoesNotExist(
+            'No ancestor of type {} could be found for this location'.format(type_code))
```

Comment:
`get_ancestors()` returns a Django queryset, so instead of this, you could just use:
```python
location.get_ancestors().get(location_type__code=type_code)
```

Reply:
oh, and @millerdev I don't think this would cause any issues, but figured you'd wanna be looped in on in-the-works ancestors query related stuff.

Reply:
Thanks! Updated